// @darwinia/types/interfaces/definitions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "balances", {
  enumerable: true,
  get: function () {
    return _definitions.default;
  }
});
Object.defineProperty(exports, "bridges", {
  enumerable: true,
  get: function () {
    return _definitions2.default;
  }
});
Object.defineProperty(exports, "bsc", {
  enumerable: true,
  get: function () {
    return _definitions3.default;
  }
});
Object.defineProperty(exports, "darwiniaInject", {
  enumerable: true,
  get: function () {
    return _definitions4.default;
  }
});
Object.defineProperty(exports, "fee", {
  enumerable: true,
  get: function () {
    return _definitions5.default;
  }
});
Object.defineProperty(exports, "headerMMR", {
  enumerable: true,
  get: function () {
    return _definitions6.default;
  }
});
Object.defineProperty(exports, "proxy", {
  enumerable: true,
  get: function () {
    return _definitions7.default;
  }
});
Object.defineProperty(exports, "relayerGame", {
  enumerable: true,
  get: function () {
    return _definitions8.default;
  }
});
Object.defineProperty(exports, "staking", {
  enumerable: true,
  get: function () {
    return _definitions9.default;
  }
});

var _definitions = _interopRequireDefault(require("./balances/definitions"));

var _definitions2 = _interopRequireDefault(require("./bridges/definitions"));

var _definitions3 = _interopRequireDefault(require("./bsc/definitions"));

var _definitions4 = _interopRequireDefault(require("./darwiniaInject/definitions"));

var _definitions5 = _interopRequireDefault(require("./fee/definitions"));

var _definitions6 = _interopRequireDefault(require("./headerMMR/definitions"));

var _definitions7 = _interopRequireDefault(require("./proxy/definitions"));

var _definitions8 = _interopRequireDefault(require("./relayerGame/definitions"));

var _definitions9 = _interopRequireDefault(require("./staking/definitions"));
